### ISSUE TITLE
ignore_changes does not work with lists of objects

### ISSUE THREAD

**Issue by bill-rich** [hashicorp/terraform#21421]
### Terraform Version
v0.12.0

### Terraform Configuration Files
```
"example_resource" "test" {
  attr1 {
    sub1 = "test1"
    sub2 = "test2"
  }
  ignore_changes = ["attr1[0].sub1"]
}
```

### Expected Behavior
Changes to the specified attribute should be ignored.

### Actual Behavior
Changes are not ignored.

### Steps to Reproduce

1. Create resource with a list and ignore one of the attributes in the list.
2. Change the ignored attribute
3. Apply

### Additional Context
`ignore_changes` works if you do not index the attribute (`ignore_changes = ["attr1"]`), but then child attributes cannot be specified.

**Comment by wgebis** [comment]
I confirm, followed `ignore_changes` option doesn't work after upgrade to 0.12:


resource "azurerm_app_service" "app-service" {
 ...
  lifecycle {
    ignore_changes = [
      "site_config[0].linux_fx_version",
    ]
  }
}

**Comment by jbardin** [comment]
Fixed by #22001
